[SEC-12866] Add documentation about events_matched template variables (#22053)
DataDog/documentation: content/en/security/notifications/variables.md

| `{{timestamp_epoch}}`                              | Time the signal was created, in milliseconds since midnight, January 1, 1970.                 |
| `{{first_seen}}`                                   | Time the signal was first seen. For example, `Mon Jan 01 00:00:00 UTC 1970`.                  |
| `{{first_seen_epoch}}`                             | Time the signal was first seen, in milliseconds since midnight, January 1, 1970.              |
| `{{last_seen}}`                                    | Time the signal was most recently triggered. For example, `Mon Jan 01 00:00:00 UTC 1970`.     |
| `{{last_seen_epoch}}`                              | Time the signal was most recently triggered, in milliseconds, since midnight, January 1, 1970.|
| `{{rule_name}}`                                    | Name of the associated rule.                                                                  |
| `{{case_name}}`                                    | Name of the triggering rule case.                                                             |

> | `{{events_matched}}`                               | Number of events which has matched the associated rule.                                       |

Reviewer: ```suggestion
| `{{events_matched}}`                               | Number of events that have matched the associated rule.                                       |
```

Author: Done in https://github.com/DataDog/documentation/pull/22053/commits/567486cdbb6ca900f898eb788d9885a54e4a98ff

> | `{{events_matched_per_query.<name_of_the_query>}}` | Number of events which has matched the associated rule query `<name_of_the_query>`.           |

Reviewer: ```suggestion
| `{{events_matched_per_query.<name_of_the_query>}}` | Number of events that have matched the associated rule query `<name_of_the_query>`.           |
```

Author: Done in https://github.com/DataDog/documentation/pull/22053/commits/567486cdbb6ca900f898eb788d9885a54e4a98ff

> When a high amount of logs match a rule, the rule title/message are not rendered for every new logs. Therefore, in this case, the rendered values of `{{events_matched}}` and `{{events_matched_per_query.<name_of_the_query>}}` could be below the values displayed in the Overview Tab of the Signal Side Panel.

Reviewer: ```suggestion
When a large number of logs match a rule, the rule's title and message are not rendered for every new log. In these cases, the rendered values of `{{events_matched}}` and `{{events_matched_per_query.<name_of_the_query>}}` could be below the values displayed in the Overview tab of the signal's side panel.
```

Author: Done in https://github.com/DataDog/documentation/pull/22053/commits/567486cdbb6ca900f898eb788d9885a54e4a98ff
